js/customize-object-selector-component.js

```diff
@@ -111,6 +111,7 @@ wp.customize.ObjectSelectorComponent = (function( api, $ ) {
 						return $.trim( component.select2_result_template( data ) );
 					},
 					templateSelection: function( data ) {
+						data.multiple = component.select2_options.multiple;
```

Comment:
It doesn't look like. I checked `this` and if there are any other arguments, and I see none. So yeah, looks good.